omics/uriparse/uri_parse.py

```diff
@@ -14,45 +14,30 @@ class OmicsUri:
     ex: omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/readSet/5346184667/source1
     ex: omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/reference/5346184667/source
     """
-
-    SCHEME = "omics"
-    NETLOC = "STORAGE"
-    DEFAULT_FILE_NAME = None
+    URI_REGEX = r"omics://(\d{10,12})\.storage\.([a-z]{2}-[a-z-]{4,}-\d+)\.amazonaws\.com/(\d{10,36})/(readSet|reference)/(\d{10,36})(/(source[12]?|index))?$"
```

Comment:
According to a quick check on the regex - looks like the slashes should be escaped?